libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -744,90 +729,55 @@ void AP_CRSF_Telem::process_vtx_telem_frame(VTXTelemetryFrame* vtx)
+void AP_CRSF_Telem::process_device_info_frame(AP_CRSF_Protocol::ParameterDeviceInfoFrame* info)
 {
     debug("process_device_info_frame: 0x%x -> 0x%x", info->origin, info->destination);
-    if (info->destination != 0 && info->destination != AP_RCProtocol_CRSF::CRSF_ADDRESS_FLIGHT_CONTROLLER) {
-        return; // request was not for us
-    }
 
-    // we are only interested in RC device info for firmware version detection
-    if (info->origin != 0 && info->origin != AP_RCProtocol_CRSF::CRSF_ADDRESS_CRSF_RECEIVER) {
+    if (!AP_CRSF_Protocol::process_device_info_frame(info, &_crsf_version, false)) {
         return;
     }
-    /*
-        Payload size is 58:
-        char[] Device name ( Null-terminated string, max len is 42 )
-        uint32_t Serial number
-        uint32_t Hardware ID
-        uint32_t Firmware ID (0x00:0x00:0xAA:0xBB AA=major, BB=minor)
-        uint8_t Parameters count
-        uint8_t Parameter version number
-    */
-    // get the terminator of the device name string
-    const uint8_t offset = strnlen((char*)info->payload,42U);
-    if (strncmp((char*)info->payload, "Tracer", 6) == 0) {
-        _crsf_version.protocol = AP_RCProtocol_CRSF::ProtocolType::PROTOCOL_TRACER;
-    } else if (strncmp((char*)&info->payload[offset+1], "ELRS", 4) == 0) {
-        // ELRS magic number is ELRS encoded in the serial number
-        // 0x45 'E' 0x4C 'L' 0x52 'R' 0x53 'S'
-        _crsf_version.protocol = AP_RCProtocol_CRSF::ProtocolType::PROTOCOL_ELRS;
-    }
-
-    if (!is_elrs()) {
-        /*
-            fw major ver = offset + terminator (8bits) + serial (32bits) + hw id (32bits) + 3rd byte of sw id = 11bytes
-            fw minor ver = offset + terminator (8bits) + serial (32bits) + hw id (32bits) + 4th byte of sw id = 12bytes
-        */
-        _crsf_version.major = info->payload[offset+11];
-        _crsf_version.minor = info->payload[offset+12];
-    } else {
-        // ELRS does not populate the version field so cook up something sensible
-        _crsf_version.major = 1;
-        _crsf_version.minor = 0;
-    }
-
-    // should we use rf_mode reported by link statistics?
-    if (is_elrs() || (!is_tracer() && (_crsf_version.major > 3 || (_crsf_version.major == 3 && _crsf_version.minor >= 72)))) {
-        _crsf_version.use_rf_mode = true;
-    }
 
     _crsf_version.pending = false;
     disable_scheduler_entry(VERSION_PING);
 }
 
 // request for a general command
-void AP_CRSF_Telem::process_command_frame(CommandFrame* command)
+void AP_CRSF_Telem::process_command_frame(AP_CRSF_Protocol::CommandFrame* command)
 {
     debug("process_command_frame: 0x%x -> 0x%x: 0x%x", command->origin, command->destination, command->payload[0]);
-    if (command->destination != 0 && command->destination != AP_RCProtocol_CRSF::CRSF_ADDRESS_FLIGHT_CONTROLLER) {
+    if (command->destination != 0 && command->destination != DeviceAddress::CRSF_ADDRESS_FLIGHT_CONTROLLER) {
```

Comment:
```suggestion
    if (command->destination != 0 && command->destination != DeviceAddress::FLIGHT_CONTROLLER) {
```

(and these should be `enum class`)